packages/blobs/src/util.ts

```diff
@@ -63,3 +66,16 @@ export function encodeName(string: string): string {
 export function decodeName(string: string): string {
   return process.platform == 'win32' ? decodeWin32SafeName(string) : string
 }
+
+// Allow users to pass in their own active span or defaults to creating a new active span
+export function withSpan<F extends (span?: Span) => ReturnType<F>>(span: Span | undefined, name: string, fn: F) {
+  if (span) {
+    // Standardize the name of the user's span with our local blob operations
+    span.updateName(name)
```

Comment:
On the fence still about this part because it forces a change onto consumers, but on the other hand it let's us answer questions of "how long does this take on average" more easily in aggregate